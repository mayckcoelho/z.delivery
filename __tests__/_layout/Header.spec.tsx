import 'jsdom-global/register'
import React from 'react'
import '../../config/setup'
import Header from '../../src/containers/_layout/header'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { unmountComponentAtNode } from 'react-dom'

let container: HTMLDivElement | null = document.createElement('div')

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: jest.fn().mockReturnValue({
    pathname: '/products',
    search: '',
    hash: '',
    state: null,
  }),
}))

beforeEach(() => {
  container && document.body.appendChild(container)
})

afterEach(() => {
  container && unmountComponentAtNode(container)
  container?.remove()
  container = null
})

describe('Header should render correctly', () => {
  it('snapshot match', () => {
    const header = shallow(<Header />)

    expect(toJson(header)).toMatchSnapshot()
  })

  it('should mount component', () => {
    mount(<Header />)
  })
})
