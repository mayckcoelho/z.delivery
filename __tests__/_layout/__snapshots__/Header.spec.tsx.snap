// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly snapshot match 1`] = `
<header>
  <div
    className="header__wrapper"
  >
    <img
      alt="Z Delivery"
      src="test-file-stub"
    />
    <div
      className="header__search"
    >
      <InputSearch
        handleClick={[Function]}
        placeholder="Pesquisar..."
      />
    </div>
  </div>
</header>
`;
